command/agent/local.go

```diff
@@ -379,6 +379,12 @@ func (l *localState) setSyncState() error {
 		}
 
 		// If our definition is different, we need to update it
+		if existing.EnableTagDrift {
+			l.logger.Printf("[DEBUG] Tag drift enabled.")
+			existing.Tags = service.Tags
```

Comment:
@ryanuber I don't think I explained myself very well re-reading what I wrote :-)

I was thinking that it's important we do the copy here because if `DeepEqual` detects a change after that (a real change we want to sync), we want the tags from the server going back up to the catalog, not the local ones as if we had the compare function. I think if we didn't copy then we'd revert the drifted tags once the service changes for any other reason.